### ISSUE TITLE
关于scan++有以下问题

### ISSUE THREAD

**Issue by Hoantrbl** [CityU-AIM-Group/SCAN#2]
1. 将batchsize设置为1，在3090（24G）上面仍然报错显存不够，跟scan显存占用完全不是一个量级（bz=4 只占用12G），非常反常，想询问程序运行具体配置情况

![image](https://user-images.githubusercontent.com/48271804/203915367-36bb7299-4bc9-4735-9243-64c832de4d36.png)

2. condagraph.py文件有报错，请问作者是如何处理这种报错？

本机配置：torch=1.10，torchvision=0.2.1

a. logits的维度长度是3，而target的维度长度是1，输入到loss里面，没有办法正确运行，我进行以下处理：

![image](https://user-images.githubusercontent.com/48271804/203915731-9df9d57f-75f1-4be4-80a7-b3bef635ec02.png)

b. 在scan++中相比于scan，这个函数增添了类似于version这样的变量，查询该函数之后并没有这个变量的条件判定，是否存在遗漏文件的情况？我进行了以下处理：

![image](https://user-images.githubusercontent.com/48271804/203915980-6631cec1-f63a-4c08-83bd-7ac598e3a545.png)

c. `agg_points`维度为3，索引后无法输入到prototype_buffer_batch中

![image](https://user-images.githubusercontent.com/48271804/203916040-b4d27801-3d99-4a2c-8554-a2ec62c3afa2.png)

3. loss文件中存在变量未设置的情况

![image](https://user-images.githubusercontent.com/48271804/203916422-60487417-1ae7-4b18-9e3e-0a0b2a774eed.png)

`self.cfg.MODEL.MIDDLE_HEAD.ACT_LOSS`在default.py或者yaml文件中并没有查询到，我修改为了`ACT_LOSS_WEIGHT`。

**Comment by Hoantrbl** [comment]
@wymanCV 此前sigma以及scan都成功运行，烦请作者核查一下该库是否能正确运行！辛苦啦

**Comment by wymanCV** [comment]
您好，感谢您对于我们这一系列工作的关注！

非常抱歉我们遗漏了两个文件的上传（相较于SCAN的版本做了一些细微的调整），分别是[loss.py](https://github.com/CityU-AIM-Group/SCAN/blob/SCAN%2B%2B/fcos_core/modeling/rpn/fcos/loss.py) 和 [transformer.py](https://github.com/CityU-AIM-Group/SCAN/blob/SCAN%2B%2B/fcos_core/layers/transformer.py)。您可以直接用我们刚最新的这两个py文件进行替换。

另外，其实SCAN(++)的DA框架有些实现细节比较次优（比如一些 梯度反传保留计算图的操作），更建议您基于SIGMA的框架~

**Comment by Hoantrbl** [comment]
> 您好，感谢您对于我们这一系列工作的关注！
> 
> 非常抱歉我们遗漏了两个文件的上传（相较于SCAN的版本做了一些细微的调整），分别是[loss.py](https://github.com/CityU-AIM-Group/SCAN/blob/SCAN%2B%2B/fcos_core/modeling/rpn/fcos/loss.py) 和 [transformer.py](https://github.com/CityU-AIM-Group/SCAN/blob/SCAN%2B%2B/fcos_core/layers/transformer.py)。您可以直接用我们刚最新的这两个py文件进行替换。
> 
> 另外，其实SCAN(++)的DA框架有些实现细节比较次优（比如一些 梯度反传保留计算图的操作），更建议您基于SIGMA的框架~

感谢您的及时回复！经文件修正后，已顺利运行！您在DA一系列工作以及HTD，对我接下来工作启发很大，非常感谢！

**Comment by Hoantrbl** [comment]
@wymanCV 

核查后发现在模型推理阶段，有个小错误：

路径为：

`fcos_core/engine/validation.py`

![image](https://user-images.githubusercontent.com/48271804/203928178-e7a9ae50-0fe2-47b0-9d88-c305e5fff062.png)

middle head函数返回为三个变量，而您的代码返回了四个。

此处修正后，代码顺利通过训练跟测试！非常感谢您的跟进！

**Comment by wymanCV** [comment]
谢谢指正，最近开源代码的时候是clone原版SCAN直接改动的，由于时间比较久远，有些文件的小调整给遗忘了，没能及时更新上。感谢~
